src/models/eventsFactory.js

```diff
@@ -296,16 +300,32 @@ class EventsFactory extends Factory {
      * Convert UTC midnight to midnights in user's timezone
      */
     dailyEvents = dailyEvents.map((item) => {
+      const groupingTimestamp = new Date(item.groupingTimestamp * 1000);
+
+      groupingTimestamp.setTime(groupingTimestamp.getTime() + timezoneOffset * 60 * 1000);
```

Comment:
why the `getMidnightWithTimezoneOffset` util is replaced with such a line?

Reply:
Because it didn't work as expected and is too complicated.

groupingTimestamp is already UTC midnight, so we just need to recalculate it with timezone offset

Reply:
looks suspicious. It was working fine and handled edge points with errors that happened near midnight. 

Why do you add a `timezoneOffset` to the Midnight? It won't be midnight after that. 

As I can see, the `getMidnightWithTimezoneOffset` has a comment describing a problem. What exactly does not work as expected?